matcher.go

```diff
@@ -47,6 +93,48 @@ func MatchRegexp(r *regexp.Regexp) MatcherFunc {
 	}
 }
 
+// MatchTimeStamp returns a MatcherFunc that matches a time layout pattern in given string
+func MatchTimeStamp(layout string) MatcherFunc{
+	return func(str string) Match {
+		var r *regexp.Regexp
+
+		switch layout {
```

Comment:
We can make a simple map lookup instead of this long switch case maybe?

```
// can be global since its constant
var layoutRegexps = map[string]*regexp.Regexp{
    time.ANSIC: ANSICRegex,
    ...
}

r := layoutRegexps[layout]
```

Reply:
Sounds good